document-service-api/src/api/openapi.yaml

```diff
@@ -0,0 +1,205 @@
+---
+openapi: 3.0.0
+info:
+  title: Appeals Document Service API Documentation
+  version: 1.0.0
+  license:
+    name: www.foundry4.com
+servers:
+  - url: http://localhost:3001/
+paths:
+  "/api/v1/{applicationId}":
```

Comment:
Can the applicationId contain '/' characters i.e. forward slashes?

Reply:
It should be a UUID. I guess is could contain slashes if it was URL encoded. This application just passes in what it receives for logical separation in the blob storage - there's no validation on the appId at all

Reply:
So is this the Appeal Id rather than the original application id? I am not sure we will have access to the original application id as a uuid. I think if we have a value that could include forward slashes as a path variable then e may have problems.
<img width="1580" alt="Screenshot 2020-12-14 at 14 27 27" src="https://user-images.githubusercontent.com/56199648/102094093-d09d3400-3e19-11eb-9462-38dc2af67075.png">

Reply:
My expectation is that it's the UUID that generated by the Forms Web App and stored in the session to link everything together.

Reply:
It seems to work in Postman so it might work from the client app that does not attempt to escape characters. 
<img width="1411" alt="Screenshot 2020-12-14 at 14 42 18" src="https://user-images.githubusercontent.com/56199648/102094735-92ecdb00-3e1a-11eb-958e-bbcdea7fe257.png">

Reply:
This is a non-issue. This should be the UUID generated by the application. That's just alphanumeric

Reply:
I suppose a deeper question is, "Does the Document Service need to know about the planning application id or appeal id at all?" If that information is provided as a linking key then we can retrieve all related docs together but in the context of our current project the list of document ids for a particular appeal is available in the appeal entity. We could acheive the same result by calling Document Service with a collection of document ids.

Reply:
Thanks for the clarification. Perhaps we could update the wording on the description and enforce that the path variable must be a uuid.

Reply:
That's exactly the point I made above. It **DOESN'T** have to be anything. There's zero validation on this, it's just used as a logical separator and used to group files. But the Doc Service has no knowledge or interest in what that value is.

This may require further work in the future on validation, but the purposes of MVP it's met the good-enough-is-good-enough stage

Reply:
If the applicationId is a uuid generated by FWA and stored in the session but not the appeal id (that is generated by the Appeals Service) then I don't think it will be persisted so may not be available when the session closes.

Reply:
It should be persisted to the Appeals Service document - I don't know about how you changed it when you did #185, but that was the purposes of the `uuid` in the original model. In any case, any documents added as part of an appeal will store the document ID (generated here as `id`, but also a UUID) and put those in the Appeals Service document.

Reply:
The appeal id is persisted as the primary key.

Reply:
Yeah, I don't care how it's stored so long as there's a unique ID in the Appeals Service. This is what should be passed in here